drivers/mcp2515/candev_mcp2515.c

```diff
@@ -444,13 +444,6 @@ static int _set_filter(candev_t *dev, const struct can_filter *filter)
         return -1;
     }
 
-    if ((f.can_id & CAN_EFF_FLAG) == CAN_EFF_FLAG) {
-        f.can_mask &= CAN_EFF_MASK;
-    }
-    else {
-        f.can_mask &= CAN_SFF_MASK;
-    }
```

Comment:
@vincent-d why was this needed in the first place?
The mask is set statically by the user, why should the driver modify it?

Reply:
The MSB of the ID contains the CAN format IIRC. 

I wonder, what behavioral difference was observed before and after removing this block?

Reply:
This block is resetting the MSB, and when setting the filter or the mast another check on the MSB (to define whether it's extended or standard) is done. But as the MSB is already reset here, any later check will be considering the CAN ID standard and thus the filter is not well applied.